File: app/src/main/java/com/example/calculator/PosishionHelper.java
package com.example.calculator;

public class PosishionHelper implements Comparable {
    public String operation;
    public int priorety;
    public int posishion;





    PosishionHelper(String operation, int posishion) {
        this.operation = operation;
        switch (operation) {
            case "+":
                priorety = 3;
                break;
            case "-":
                priorety = 3;
                break;
            case "\u00d7":
                priorety = 2;
                break;
            case "/":
                priorety = 2;
                break;
            case "^":
                priorety = 1;
                break;
        }
        this.posishion = posishion;
    }


    @Override
    public int compareTo(Object o) {
        return this.priorety - ((PosishionHelper) o).priorety;
    }

    @Override
    public String toString() {
        return operation + " позиция первого элемента: " + posishion;
    }

}

File: app/src/main/java/com/example/calculator/PosishionHelper.kt
package com.example.calculator

class PosishionHelper internal constructor(var operation: String, posishion: Int) :
    Comparable<Any?> {
    var priorety = 0
    var posishion: Int

    init {
        when (operation) {
            "+" -> priorety = 3
            "-" -> priorety = 3
            "\u00d7" -> priorety = 2
            "/" -> priorety = 2
            "^" -> priorety = 1
        }
        this.posishion = posishion
    }

    override fun compareTo(o: Any?): Int {
        return priorety - (o as PosishionHelper?)!!.priorety
    }

    override fun toString(): String {
        return "$operation позиция первого элемента: $posishion"
    }
}